### PR TITLE
[Feature] CON 81 - Enable text wrap passages

### PR DESCRIPTION
**Pull request by habemuscode**
**Related to this issue**

https://oat-sa.atlassian.net/browse/CON-81

**Other related PRs**

- https://github.com/oat-sa/tao-core-ui-fe/pull/217
- https://github.com/oat-sa/extension-tao-itemqti/pull/1580

**Story**

As the author, I want to be able to wrap my text around pictures, so that I can produce passages that look like real-life media content.

Currently, I am only able to anchor images AS a character. In order for the text to be able to flow around it, I will need to be able to anchor it TO a character and specify how text is supposed to wrap around it

Required wrap modes:

- Float off
- Float right
- Float left

**Acceptance Criteria**

- Anchoring an image as a character should still be possible
- Text wrapping should not be available when the image is anchored as a character
- Images maintain position within their paragraph when text is added before or after
- Responsive images are still supported if the image is anchored to a character
- Exported items will contain the relevant CSS to ensure the rendering will be correct in any environment
- Exporting and reimporting an item will maintain the anchoring and wrapping

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -28,7 +28,7 @@
     'label' => 'extension-tao-mediamanager',
     'description' => 'TAO media manager extension',
     'license' => 'GPL-2.0',
-    'version' => '11.6.1',
+    'version' => '11.7.0',
     'author' => 'Open Assessment Technologies SA',
     'requires' => [
         'tao' => '>=46.6.0',
```
Comment (gabrielfs7) [inline manifest.php:34]: We need to update tao-core version here too :)

**Review by gabrielfs7 — APPROVED** [review]
LGTM BE code, just remember to update the manifest with tao-core before merge :) 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: views/js/qtiCreator/widgets/static/img/states/Active.js [hunk 2]
```diff
@@ -76,19 +76,18 @@ define([
             .replace(/\W/, ' ')
             .substr(0, 255);
     };
-
     const _initAlign = function (widget) {
         let align = 'default';
+        const className = widget.$original[0].classList;
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/static/img/states/Active.js:81]: Why `widget.$original[0]` and not `widget.element`?

File: views/js/qtiCreator/widgets/static/img/states/Active.js [hunk 3]
```diff
@@ -148,11 +147,23 @@ define([
...
+                        active: true
                     }
                 };
                 media.$container = $mediaSpan.parents('.widget-box');
                 mediaEditor = mediaEditorComponent($mediaResizer, media, options).on('change', function (nMedia) {
                     media = nMedia;
+                    media.$container.css('min-height', $img.height() + 16 + 'px');
+                    inlineHelper.positionFloat(widget, media.align);
+                    if (media.align === 'right') {
+                        widget.$container.css('margin', '0 0 0 16px');
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/static/img/states/Active.js:161]: We should't put inline css in XML, can you use css class for this?

**Review by bziondik — CHANGES_REQUESTED** [review]
No inline css styles in XML, better to use class.

File: views/js/qtiCreator/widgets/static/img/states/Active.js (hunk 3 above)
Comment (habemuscode) [reply views/js/qtiCreator/widgets/static/img/states/Active.js]: Done

File: views/js/qtiCreator/widgets/static/img/states/Active.js (hunk 2 above)
Comment (habemuscode) [reply views/js/qtiCreator/widgets/static/img/states/Active.js:81]: imagen

**Review by bziondik — CHANGES_REQUESTED** [review]
Works for CurGen TR, but doesn't work for TR new UI
image
image
image

**Review by 404passionFruit — APPROVED** [review]
Please provide the ticket for New UI TR team and we're good to go

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by 404passionFruit** [comment]
Converted to draft as there is an alternative solution https://github.com/oat-sa/extension-tao-mediamanager/pull/277

**Comment by habemuscode** [comment]
Closed in favor of https://github.com/oat-sa/extension-tao-mediamanager/pull/277

**Pull request closed without merging**